src/modules/expression/binop/add.rs

```diff
@@ -68,6 +68,12 @@ impl TranslateModule for Add {
                 let id = meta.gen_array_id();
                 let name = format!("__AMBER_ARRAY_ADD_{id}");
                 meta.stmt_queue.push_back(format!("{name}=({left} {right})"));
+                if let Some(append) = self.left.append_let(meta, &name, false) {
```

Comment:
This function has a bug where, if a `lines` expression and an existing array are concatenated, the entries end up in the wrong order.  This is exhibited by the following script:

```
hwalters@Ghostwheel ~/git/amber-lang (implement-builtin-lines-function-ii)
$ cat 1-2-3.txt
1
2
3
```
```
hwalters@Ghostwheel ~/git/amber-lang (implement-builtin-lines-function-ii)
$ cat concat.ab
#!/usr/bin/env amber
for line in lines("1-2-3.txt") + ["4", "5", "6"] {
    echo line
}
```
```
hwalters@Ghostwheel ~/git/amber-lang (implement-builtin-lines-function-ii)
$ ./concat.ab
4
5
6
1
2
3
```
We should probably fix that.